sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs

```diff
@@ -1,11 +1,47 @@
 namespace Azure.Data.Tables
 {
+    public partial class DynamicTableEntity : System.Collections.Generic.ICollection<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IDictionary<string, object>, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.IEnumerable
+    {
+        public DynamicTableEntity() { }
+        public DynamicTableEntity(System.Collections.Generic.IDictionary<string, object> values) { }
+        public DynamicTableEntity(string partitionKey, string rowKey) { }
+        public int Count { get { throw null; } }
+        public string ETag { get { throw null; } set { } }
+        public bool IsReadOnly { get { throw null; } }
+        public object this[string key] { get { throw null; } set { } }
+        public System.Collections.Generic.ICollection<string> Keys { get { throw null; } }
+        public string PartitionKey { get { throw null; } set { } }
+        public string RowKey { get { throw null; } set { } }
+        public System.DateTimeOffset Timestamp { get { throw null; } }
+        public System.Collections.Generic.ICollection<object> Values { get { throw null; } }
+        public void Add(System.Collections.Generic.KeyValuePair<string, object> item) { }
+        public void Add(string key, object value) { }
+        public void Clear() { }
+        public bool Contains(System.Collections.Generic.KeyValuePair<string, object> item) { throw null; }
+        public bool ContainsKey(string key) { throw null; }
+        public void CopyTo(System.Collections.Generic.KeyValuePair<string, object>[] array, int arrayIndex) { }
+        public byte[] GetBinary(string key) { throw null; }
+        public bool GetBoolean(string key) { throw null; }
+        public System.DateTime GetDateTime(string key) { throw null; }
+        public double GetDouble(string key) { throw null; }
+        public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, object>> GetEnumerator() { throw null; }
+        public System.Guid GetGuid(string key) { throw null; }
+        public int GetInt32(string key) { throw null; }
+        public long GetInt64(string key) { throw null; }
+        public string GetString(string key) { throw null; }
+        public bool Remove(System.Collections.Generic.KeyValuePair<string, object> item) { throw null; }
```

Comment:
Most folks won't be using the full gamut of dictionary features directly and the API gets a little cluttered.  We should implement a lot of the `IDictionary` members explicitly.  You can see https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation for how that works or https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/SearchDocument/SearchDocument.Dictionary.cs for how we did that in Search.

Reply:
tracking here #13658